lib/active_merchant/billing/gateways/payu_latam.rb

```diff
@@ -0,0 +1,484 @@
+
+      QUERY_TEST_URL = "https://sandbox.api.payulatam.com/reports-api/4.0/service.cgi"
+      QUERY_LIVE_URL = "https://api.payulatam.com/reports-api/4.0/service.cgi"
+
+      BRAND_MAP = {
+        "visa" => "VISA",
+        "master" => "MASTERCARD",
+        "american_express" => "AMEX",
+        "diners_club" => "DINERS"
+      }
+
+      def initialize(options={})
+        requires!(options, :merchant_id, :account_id, :api_login, :api_key)
+        super
+      end
+
+      def purchase(amount, payment_method, options={})
+        post = {}
+        auth_or_sale(post, 'AUTHORIZATION_AND_CAPTURE', amount, payment_method, options)
+        commit('purchase', post)
+      end
+
+      def authorize(amount, payment_method, options={})
+        post = {}
+        auth_or_sale(post, 'AUTHORIZATION', amount, payment_method, options)
+        commit('auth', post)
+      end
+
+      def capture(authorization, options={})
+        post = {}
+
+        add_credentials(post, 'SUBMIT_TRANSACTION')
+        add_transaction_type(post, 'CAPTURE')
+        add_reference(post, authorization)
+
+        commit('capture', post)
+      end
+
+      def void(authorization, options={})
+        post = {}
+
+        action, _, _ = split_authorization(authorization)
+
+        case action
+        when 'auth'
+          add_credentials(post, 'SUBMIT_TRANSACTION')
+          add_transaction_type(post, 'VOID')
+          add_reference(post, authorization)
+        else
+          message = "Unsupported operation: successful authorize transactions can only be voided."
+          return Response.new(false, message)
+        end
+
+        commit('void', post)
+      end
+
+      def refund(authorization, options={})
+        post = {}
+
+        action, _, _ = split_authorization(authorization)
+
+        case action
+        when 'purchase', 'capture'
+          add_credentials(post, 'SUBMIT_TRANSACTION')
+          add_transaction_type(post, 'REFUND')
+          add_reference(post, authorization)
+        else
+          message = "Unsupported operation: successful purchase and capture transactions can only be refunded."
+          return Response.new(false, message)
+        end
+
+        commit('refund', post)
+      end
+
+      # Supported countries have different minimum amounts and
+      # verify uses the least valid amount across all countries
+      def verify(credit_card, options={})
+        MultiResponse.run(:use_first_response) do |r|
+          r.process { authorize(4000, credit_card, options) }
+          r.process(:ignore_result) { void(r.authorization, options) }
+        end
+      end
+
+      def store(payment_method, options = {})
+        post = {}
+
+        add_credentials(post, 'CREATE_TOKEN')
+        add_payment_method_to_be_tokenized(post, payment_method)
+
+        commit('store', post)
+      end
+
+      def verify_credentials
+        response = authorize(4000, "_|VISA|00000000-0000-0000-0000-000000000000", {cvv: '000'})
```

Comment:
That's a pretty cool idea.  

I wonder if they have an API call that could be used that doesn't take a card.  The main thought I had was would we break if they changed the message text.  Seems like historically, messages are more likely to change than response codes.  But perhaps messages are one of the only ways to distinguish between error states.  Or maybe this message won't change and we'll be fine.